lib/runtime/storage/storagediff.go

```diff
@@ -0,0 +1,230 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package storage
+
+import (
+	"bytes"
+	"maps"
+	"sync"
+
+	xmaps "golang.org/x/exp/maps"
+
+	"github.com/ChainSafe/gossamer/pkg/trie"
+	"github.com/ChainSafe/gossamer/pkg/trie/codec"
+)
+
+type storageDiff struct {
+	upserts        map[string][]byte
```

Comment:
Is upsert a typical term used in this context? I had never heard of it and was confused, but if its a typical term I am okay with it

Reply:
I got it from databases context, where you want to mix "update" + "insert" https://www.cockroachlabs.com/blog/sql-upsert/#:~:text=The%20term%20upsert%20is%20a,value%20doesn't%20already%20exist.